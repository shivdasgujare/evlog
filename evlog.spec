#
# Enterprise Event Log Spec
#
Summary:   Enterprise Event Logging
Name:      evlog
Version:   1.6.1 
Release:   1
Copyright: LGPL
Vendor:    IBM <http://www.ibm.com>
Url:       http://evlog.sourceforge.net

Packager:  Hien Nguyen <nguyhien@us.ibm.com>
Group:     System Environment/Base
Source:    evlog-1.6.1.tar.gz
BuildRoot: /var/tmp/%{name}-buildroot

%description
Enterprise Event Logging

%prep
%setup
CFLAGS="$RPM_OPT_FLAGS" CXXFLAGS="$RPM_OPT_FLAGS" $LOCALFLAGS
./configure 

%build
make -C user

%install
rm -rf $RPM_BUILD_ROOT
make DESTDIR=$RPM_BUILD_ROOT install

# Copy ela* stuff to /usr/bin for now, we will use move later.
mkdir -p $RPM_BUILD_ROOT/usr/sbin
cp $RPM_BUILD_ROOT/sbin/ela_* $RPM_BUILD_ROOT/usr/sbin

cd $RPM_BUILD_ROOT
find . -type d | sed '1,2d;s,^\.,\%attr(-\,root\,root) \%dir ,' |grep -v '/etc/init.d' > $RPM_BUILD_DIR/file.list.evlog-user
find . -type f | sed 's,^\.,\%attr(-\,root\,root) ,' >> $RPM_BUILD_DIR/file.list.evlog-user
find . -type l | sed 's,^\.,\%attr(-\,root\,root) ,' >> $RPM_BUILD_DIR/file.list.evlog-user

%pre
#echo This is pre arg=$1
if [ "$1" = "1" -o "$1" = "2" ] ; then
	if [ -f /etc/init.d/evlaction ]
	then
		PID3=`ps -ef | grep evlactiond | grep -v grep | awk '{print $2}'`
		test -z $PID3 || /etc/init.d/evlaction stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlnotify ]
	then
		PID2=`ps -ef | grep evlnotifyd | grep -v grep | awk '{print $2}'`
		test -z $PID2 || /etc/init.d/evlnotify stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlogrmtd ]
	then
		PID1=`ps -ef | grep evlogrmtd | grep -v grep | awk '{print $2}'`
		test -z $PID1 || /etc/init.d/evlogrmtd stop
		/bin/sleep 1
	fi	
	if [ -f /etc/init.d/evlog ]
	then
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		test -z $PID || /etc/init.d/evlog stop
		/bin/sleep 1
	fi	

	# We don't use logrotate anymore so remove evlog under logrotate.d 
	if [ -f /etc/logrotate.d/evlog ]
	then
		rm -f /etc/logrotate.d/evlog
	fi	
	
	if [ -d /etc/rc.d/init.d ]
	then 
		# Redhat style, make a link to /etc/init.d
		test -L /etc/init.d || ln -s /etc/rc.d/init.d /etc/init.d
	fi
	
	if [ -d /etc/init.d/rc3.d ]
	then
		# Suse style of rc0-6.d
		test -L /etc/rc.d || ln -s /etc/init.d /etc/rc.d
	fi
fi
	
%post
#echo This is post arg=$1
if [ "$1" = "1" -o "$1" = "2" ] ; then
	/sbin/ldconfig

	(cd /var/evlog/templates; make)
	
	/etc/init.d/evlnotify start
	/bin/sleep 1
	/etc/init.d/evlaction start
	/bin/sleep 1
	/etc/init.d/evlog start
	/bin/sleep 1
	/etc/init.d/evlogrmt start
	
	REL_INITDOTD_DIR=../../init.d
	RCDOTD_BASE=/etc/rc.d

	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc3.d/S09evlog
	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc5.d/S09evlog
	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc0.d/K90evlog
	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc6.d/K90evlog

	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc3.d/S89evlogrmt
	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc5.d/S89evlogrmt
	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc0.d/K60evlogrmt
	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc6.d/K60evlogrmt
	
	# Remove previous version (1.5.3 and earlier) links if exist
	test ! -L $RCDOTD_BASE/rc3.d/S80evlnotify || rm -f $RCDOTD_BASE/rc3.d/S80evlnotify
	test ! -L $RCDOTD_BASE/rc5.d/S80evlnotify || rm -f $RCDOTD_BASE/rc5.d/S80evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc3.d/S07evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc5.d/S07evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc0.d/K50evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc6.d/K50evlnotify
	
	# Remove previous version (1.5.3 and earlier) links if exist
	test ! -L $RCDOTD_BASE/rc3.d/S90evlaction || rm -f $RCDOTD_BASE/rc3.d/S90evlaction
	test ! -L $RCDOTD_BASE/rc5.d/S90evlaction || rm -f $RCDOTD_BASE/rc5.d/S90evlaction

	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc3.d/S08evlaction
	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc5.d/S08evlaction
	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc0.d/K40evlaction
	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc6.d/K40evlaction
fi


%preun
#echo This is preun arg=$1
if [ "$1" = "0" ] ; then

	if [ -f /etc/init.d/evlaction ]
	then
		PID3=`ps -ef | grep evlactiond | grep -v grep | awk '{print $2}'`
		test -z $PID3 || /etc/init.d/evlaction stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlnotify ]
	then
		PID2=`ps -ef | grep evlnotifyd | grep -v grep | awk '{print $2}'`
		test -z $PID2 || /etc/init.d/evlnotify stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlogrmt ]
	then
		PID1=`ps -ef | grep evlogrmtd | grep -v grep | awk '{print $2}'`
		test -z $PID1 || /etc/init.d/evlogrmt stop
		/bin/sleep 1
	fi	
	if [ -f /etc/init.d/evlog ]
	then
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		test -z $PID || /etc/init.d/evlog stop
		/bin/sleep 1
	fi	

	(cd /var/evlog/templates; make clobber)

	#
	# remove libevlsyslog.so.1 entry in the /etc/ld.so.preload if any
	#
	/sbin/slog_fwd -r
fi
	
%postun
#echo This is postun arg=$1
if [ "$1" = "0" ] ; then
	#
	# remove startup scripts link
	#

	if [ -e /etc/init.d/evlog ]
	then
		rm -f /etc/init.d/evlog
	fi
	if [ -e /etc/init.d/evlogrmt ]
	then
		rm -f /etc/init.d/evlogrmt
	fi
	if [ -e /etc/init.d/evlnotify ]
	then
		rm -f  /etc/init.d/evlnotify
	fi
	if [ -e /etc/init.d/evlaction ]
	then
		rm -f /etc/init.d/evlaction
	fi 
	
	RCDOTD_BASE=/etc/rc.d

	rm -f $RCDOTD_BASE/rc3.d/S09evlog
	rm -f $RCDOTD_BASE/rc5.d/S09evlog
	rm -f $RCDOTD_BASE/rc0.d/K90evlog
	rm -f $RCDOTD_BASE/rc6.d/K90evlog

	rm -f $RCDOTD_BASE/rc3.d/S89evlogrmt
	rm -f $RCDOTD_BASE/rc5.d/S89evlogrmt
	rm -f $RCDOTD_BASE/rc0.d/K60evlogrmt
	rm -f $RCDOTD_BASE/rc6.d/K60evlogrmt

	rm -f $RCDOTD_BASE/rc3.d/S07evlnotify
	rm -f $RCDOTD_BASE/rc5.d/S07evlnotify
	rm -f $RCDOTD_BASE/rc0.d/K50evlnotify
	rm -f $RCDOTD_BASE/rc6.d/K50evlnotify

	rm -f $RCDOTD_BASE/rc3.d/S08evlaction
	rm -f $RCDOTD_BASE/rc5.d/S08evlaction
	rm -f $RCDOTD_BASE/rc0.d/K40evlaction
	rm -f $RCDOTD_BASE/rc6.d/K40evlaction

	/sbin/ldconfig	
fi
	

%clean
rm -rf $RPM_BUILD_ROOT/*
rm -rf $RPM_BUILD_DIR/evlog
rm -rf ../file.list.evlog


%files -f ../file.list.evlog-user

%changelog
* Tue Apr 27 2004 Hien Nguyen <nguyhien@us.ibm.com>
- Release version of v1.6.0. Added ELA feature.
* Thu Dec 18 2003 Hien Nguyen <nguyhien@us.ibm.com>
- Added ELA feature - and bugs fixed
* Mon Jun 16 2003 Hien Nguyen <nguyhien@us.ibm.com>
- Add multi-arch support
* Thu Mar 7 2002 Jim Keniston <kenistoj@us.ibm.com>
- Added kernel module tests kfacreg and kfacreg2 (test3, test4)
* Fri Oct 12 2001 Hien Nguyen <nguyhien@us.ibm.com>
- Bug fixes, add template, evltc, evlfacility, libevlsyslog
  and many small enhancements.
* Fri Sep 28 2001 Hien Nguyen <nguyhien@us.ibm.com>
- Bug fixes, adding s390 and s390x specific tests 
* Tue Aug 26 2001 Hien Nguyen <nguyhien@us.ibm.com>
- First RPM build
