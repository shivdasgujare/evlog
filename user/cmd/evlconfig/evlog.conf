#
# Configuration values for evlogd daemon. 
# Use 'evlconfig' command to update these values. If this file is edited 
# manually, the updated value will not be used by the evlogd daemon. So,
# the evlogd daemon has to be restarted. Hence, use 'evlconfig' command
# to send this values to evlogd daemon.
# Console display level - OFF=111
#
Discard Dups: on
Discard Interval: 3
Discard Count: 100
Event Screen: nofilter
Console display level: 111
Lookback Size: 3
