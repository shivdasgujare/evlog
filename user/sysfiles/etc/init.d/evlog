#!/bin/sh
#
# chkconfig: 35 09 90
# description: Enterprise event log daemon
#
#

# Source function library.

REDHAT=`cat /proc/version | grep "Red Hat"`
# Source function library.
if [ ! -z "$REDHAT" ]
then
	. /etc/init.d/functions
fi


[ -f /sbin/evlogd ] || exit 0

EVLOGD_OPTIONS=""
# Source config
if [ -f /etc/sysconfig/evlog ]; then
	. /etc/sysconfig/evlog
fi

RETVAL=0

umask 077

record_last_recid() {
	/sbin/evlview -l $1 -t 1 -S %recid% > $2 2> /dev/null
	test -s $2 || echo 0 > $2
	chmod 644 $2
}

start() {
 	echo -n $"Starting enterprise event logger: "

	record_last_recid /var/evlog/eventlog /var/evlog/bootrecid
	record_last_recid /var/evlog/privatelog /var/evlog/pbootrecid
 	
 	if [ ! -z "$REDHAT" ]
	then 	
		daemon /sbin/evlogd $EVLOGD_OPTIONS
	else
		/sbin/evlogd $EVLOGD_OPTIONS
	fi
	sleep 1
	RETVAL=$?
	echo
	touch /var/lock/subsys/evlog
	return $RETVAL
}	
stop() {
	echo -n $"Shutting down enterprise event logger: "
	
	if [ ! -z "$REDHAT" ]
	then
		killproc evlogd
	else
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		if [ ! -z "$PID" ]
		then
			kill -s SIGTERM $PID
		fi
	fi

	RETVAL=$?
	echo
	rm -f /var/lock/subsys/evlog
	if [ -e /var/run/evlogd.pid ]
	then
		rm -f /var/run/evlogd.pid
	fi
	return $RETVAL
}
rhstatus() {
	status evlogd
}
restart() {
	stop
	start
}	

case "$1" in
  start)
  	start
	;;
  stop)
  	stop
	;;
  status)
  	rhstatus
	;;
  restart|reload)
  	restart
	;;
  condrestart)
    restart	
	;;
  *)
	echo $"Usage: $0 {start|stop|status|restart|condrestart}"
	exit 1
esac

exit $?

