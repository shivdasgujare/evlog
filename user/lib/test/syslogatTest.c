#include <stdlib.h>
#include <syslog.h>
#include <assert.h>
#include "evlog.h"

#define EVL_FACILITY_NAME LOCAL7

void
game(const char *opponent, const char *where, int our_score, int their_score,
	const char *commentary)
{
	assert(our_score != their_score);	/* No ties allowed */
	if (our_score > their_score) {
		syslogat(LOG_INFO,
"OSU beat {opponent}%s in {where}%s, {us}%d-{them}%d\n{{commentary}%s",
			opponent, where, our_score, their_score, commentary);
	} else {
		syslogat(LOG_ERR,
"{opponent}%s beat OSU in {where}%s, {them}%d-{us}%d\n{{commentary}%s",
			opponent, where, their_score, our_score, commentary);
	}
}

main()
{
	/* syslog messages to LOCAL1, evlog to LOCAL7 */
	openlog("syslogatTest", 0, LOG_LOCAL1);
	game("Eastern Kentucky", "Corvallis", 49, 10, "");
	game("Temple", "Philadelphia", 35, 3, "");
	game("UNLV", "Corvallis", 47, 17, "");
	game("Fresno St", "Corvallis", 59, 19, "Wow!  How good is this team?");
	game("USC", "LA", 0, 22, "Not great, apparently.");
	game("UCLA", "Corvallis", 35, 43, "");
	game("Arizona St", "Tempe", 9, 13, "PAC-10 looking good this year");
	game("California", "Corvallis", 24, 13, "");
	game("Arizona", "Corvallis", 38, 3, "");
	game("Washington", "Seattle", 29, 41, "No mud in Husky Stadium");
	game("Stanford", "Palo Alto", 31, 21, "Looked grim at halftime");
	game("Oregon", "Corvallis", 45, 24, "Ducks' nightmare continues");
	exit(0);
}
