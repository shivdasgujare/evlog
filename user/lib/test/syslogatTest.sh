#!/bin/sh
#
fail() {
	echo syslogat test	: FAILED
	exit 1
}

if [ ! -x /sbin/evlgentmpls ]
then
	echo "Skipping syslogat test: /sbin/evlgentmpls not installed."
	exit 0
fi

TEST_TEMPLATE=syslogatTest.out

export EVLTMPLPATH=$EVL_TESTDIR/templates

if [ "$USER" != root ]
then
	echo >&2 "syslogatTest.sh must be run as root, since it creates files"
	echo >&2 "in $EVLTMPLPATH."
	fail
fi

rm -f $EVLTMPLPATH/local7/local7.t

/sbin/evlgentmpls $EVLTMPLPATH syslogatTest
if [ $? -ne 0 ]
then
	fail
fi

/sbin/evltc $EVLTMPLPATH/local7/local7.t
if [ $? -ne 0 ]
then
	fail
fi

lastRecid=`/sbin/evlview -t 1 -S %recid%`
./syslogatTest
if [ $? -ne 0 ]
then
	fail
fi
sleep 1

/sbin/evlview -b -f "recid>$lastRecid && where" -S %data% | sed -e 's/game:.*/game:x/' > syslogatTest.sh.out
echo '-----' >> syslogatTest.sh.out
/sbin/evlview -b -f "recid>$lastRecid && where" -S '%file%:%opponent%:%where%:%us%:%them%:%commentary%' >> syslogatTest.sh.out

diff syslogatTest.sh.out $TEST_TEMPLATE

if [ $? -eq "0" ]
then
	echo syslogat test	:PASSED
else
	fail
fi
rm -f syslogatTest.sh.out
exit 0
