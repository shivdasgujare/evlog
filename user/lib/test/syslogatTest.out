syslogatTest.c:game:x
OSU beat Eastern Kentucky in Corvallis, 49-10
syslogatTest.c:game:x
OSU beat Temple in Philadelphia, 35-3
syslogatTest.c:game:x
OSU beat UNLV in Corvallis, 47-17
syslogatTest.c:game:x
OSU beat Fresno St in Corvallis, 59-19
syslogatTest.c:game:x
USC beat OSU in LA, 22-0
syslogatTest.c:game:x
UCLA beat OSU in Corvallis, 43-35
syslogatTest.c:game:x
Arizona St beat OSU in Tempe, 13-9
syslogatTest.c:game:x
OSU beat California in Corvallis, 24-13
syslogatTest.c:game:x
OSU beat Arizona in Corvallis, 38-3
syslogatTest.c:game:x
Washington beat OSU in Seattle, 41-29
syslogatTest.c:game:x
OSU beat Stanford in Palo Alto, 31-21
syslogatTest.c:game:x
OSU beat Oregon in Corvallis, 45-24
-----
syslogatTest.c:Eastern Kentucky:Corvallis:49:10:
syslogatTest.c:Temple:Philadelphia:35:3:
syslogatTest.c:UNLV:Corvallis:47:17:
syslogatTest.c:Fresno St:Corvallis:59:19:Wow!  How good is this team?
syslogatTest.c:USC:LA:0:22:Not great, apparently.
syslogatTest.c:UCLA:Corvallis:35:43:
syslogatTest.c:Arizona St:Tempe:9:13:PAC-10 looking good this year
syslogatTest.c:California:Corvallis:24:13:
syslogatTest.c:Arizona:Corvallis:38:3:
syslogatTest.c:Washington:Seattle:29:41:No mud in Husky Stadium
syslogatTest.c:Stanford:Palo Alto:31:21:Looked grim at halftime
syslogatTest.c:Oregon:Corvallis:45:24:Ducks' nightmare continues
